go-runtime/compile/schema.go

```diff
@@ -461,12 +482,56 @@ func visitGenDecl(pctx *parseContext, node *ast.GenDecl) {
 	}
 }
 
+func maybeVisitTypeEnumVariant(pctx *parseContext, node *ast.GenDecl, directives []directive) bool {
+	if len(node.Specs) != 1 {
+		return false
+	}
+	// If any directives on this node are exported, then the enum variant node is
+	// considered exported. Also, if the parent enum itself is exported, then all its
+	// variants should transitively also be exported.
+	isExported := false
```

Comment:
i think the behavior should be slightly different in that we should by default inherit the parent enum visibility, but overwrite it if the variant itself has an explicit visibility/export directive. that way the explicit annotation takes precedence.

one way to accomplish-
```suggestion
	isExported := pctx.enums[enumName].Export
```

and below always overwriting if we find an export annotation (`exportableDir.IsExported()` may return false, even when `pctx.enums[enumName].Export` is true, and we should respect that):
```
if exportableDir, ok := dir.(exportable); ok {
  isExported = exportableDir.IsExported()
}
```

Reply:
actually better yet, if `pctx.enums[enumName].Export` is true and `exportableDir.IsExported()` is false, we can add an error on that variant node describing that it must be exported to be a variant of enum X, and exit

Reply:
That's a really good point. I'll make that change now and add a test for the error as well.